src/components/subtitlesync/subtitlesync.js

```diff
@@ -1,125 +1,138 @@
 
-    subtitleSyncTextField.blur = function () {
-        // prevent textfield to blur while element has focus
-        if (!this.hasFocus && this.prototype) {
-            this.prototype.blur();
-        }
-    };
+        this._initSlider();
+        this._initTextField();
 
-    function updateSubtitleOffset() {
-        const value = parseFloat(subtitleSyncSlider.value);
-        // set new offset
-        playbackManager.setSubtitleOffset(value, player);
-        // synchronize with textField value
-        subtitleSyncTextField.updateOffset(value);
+        // Set initial offset
+        this.reset();
     }
 
-    subtitleSyncSlider.updateOffset = function (sliderValue) {
-        // default value is 0s = 0ms
-        this.value = sliderValue === undefined ? 0 : sliderValue;
+    get currentOffset() {
+        return parseFloat(this.slider.value);
+    }
 
-        updateSubtitleOffset();
-    };
+    set currentOffset(value) {
+        this.slider.value = value.toString();
+        playbackManager.setSubtitleOffset(value, this.player);
+        this.textField.updateOffset(value);
+    }
 
-    subtitleSyncSlider.addEventListener('change', () => updateSubtitleOffset());
+    _initSlider() {
+        const slider = this.slider;
 
-    subtitleSyncSlider.getBubbleHtml = function (_, value) {
-        return '<h1 class="sliderBubbleText">'
-            + (value > 0 ? '+' : '') + parseFloat(value) + 's'
-            + '</h1>';
-    };
+        if (layoutManager.tv) {
+            slider.classList.add('focusable');
+            // eslint-disable-next-line no-warning-comments
+            // HACK: Delay to give time for registered element attach (Firefox)
+            setTimeout(() => slider.enableKeyboardDragging(), 0);
+        }
+
+        // When slider changes we assign the value to the currentOffset to trigger the setter
+        slider.addEventListener('change', () => {
+            // eslint-disable-next-line no-self-assign
+            this.currentOffset = this.currentOffset;
+        });
+
+        slider.getBubbleHtml = createSliderBubbleHtml;
+    }
+
+    _initTextField() {
+        const textField = this.textField;
+
+        textField.updateOffset = (offset) => {
+            textField.textContent = offset + 's';
+        };
+
+        textField.addEventListener('click', () => {
+            // keep focus to prevent fade with osd
+            textField.hasFocus = true;
+        });
+
+        textField.addEventListener('keydown', (event) => {
+            if (event.key === 'Enter') {
+                // if input key is enter search for float pattern
+                let inputOffset = /[-+]?\d+\.?\d*/g.exec(textField.textContent);
+                if (inputOffset) {
+                    inputOffset = parseFloat(inputOffset[0]);
+                    this.currentOffset = inputOffset;
+                } else {
+                    textField.textContent = (playbackManager.getPlayerSubtitleOffset(this.player) || 0) + 's';
```

Comment:
```suggestion
                    textField.updateOffset(playbackManager.getPlayerSubtitleOffset(this.player) || 0);
```

In the next PR, we need to introduce a new translatable string for offset and use it in `textField.updateOffset`.